### PR TITLE
New composer - add audio comp

### LINKED ISSUE #15729: Reimplement composer - add audio component

### PR DESCRIPTION
**Pull request by OmarBasem**
fixes: https://github.com/status-im/status-mobile/issues/15729

### CODE REVIEW

File: src/status_im2/contexts/chat/bottom_sheet_composer/actions/view.cljs [hunk 1]
```diff
@@ -64,13 +64,57 @@
...
+  [{:keys [record-permission? audio-file]}
+   {:keys [recording? gesture-enabled?]}
+   {:keys [container-opacity]}]
+  [rn/view
+   {:style          (style/record-audio-container)
+    :pointer-events :box-none}
+   [quo/record-audio
+    {:record-audio-permission-granted    record-permission?
+     :on-start-recording                 (fn []
+                                           (reset! recording? true)
```
Comment (OmarBasem) [inline src/status_im2/contexts/chat/bottom_sheet_composer/actions/view.cljs]: @briansztamfater I am not sure what was `on-init` function doing? I tried removing it and did not notice any issue. Please let me know if it is really needed.

**Comment by OmarBasem** [comment]
Hi @briansztamfater,
when adding the audio component to the new composer I noticed an issue that happens on re-entering the chat screen after recording an audio. The send and delete buttons are not displayed. This issue is also happening on the old composer on `develop`. I tried to debug the issue but the `quo2` audio component is a bit complex. Are you aware of that issue or tracking it somewhere?

File: src/status_im2/contexts/chat/bottom_sheet_composer/actions/view.cljs (hunk 1 above)
Comment (briansztamfater) [reply src/status_im2/contexts/chat/bottom_sheet_composer/actions/view.cljs]: It is used to save the reset function in `record-audio-reset-fn` atom from the audio component and reset the state on show-send / hide-send functions in current (old) composer implementation. One of the reasons, for example, let's say a user is replying to a message and recording an audio while the keyboard is being shown. Then, in the middle of the recording, the user starts typing a message, causing the the recorder to be hidden and text with the send button to be shown, so `@record-audio-reset-fn` is called to stop the recording and reset current audio component to its initial state.

**Comment by briansztamfater** [comment]
> Hi @briansztamfater, when adding the audio component to the new composer I noticed an issue that happens on re-entering the chat screen after recording an audio. The send and delete buttons are not displayed. This issue is also happening on the old composer on `develop`. I tried to debug the issue but the `quo2` audio component is a bit complex. Are you aware of that issue or tracking it somewhere?

Not that I am aware of, I tested those cases carefully while initially integrating the component so maybe something broke, I will check.

**Comment by briansztamfater** [comment]
@OmarBasem Please cherry-pick [e4d7b2828](https://github.com/status-im/status-mobile/commit/e4d7b2828) and let me know if that fixes the issue

File: src/status_im2/contexts/chat/bottom_sheet_composer/actions/view.cljs (hunk 1 above)
Comment (OmarBasem) [reply src/status_im2/contexts/chat/bottom_sheet_composer/actions/view.cljs]: Got it! Thank you @briansztamfater

**Comment by OmarBasem** [comment]
> @OmarBasem Please cherry-pick [e4d7b2828](https://github.com/status-im/status-mobile/commit/e4d7b2828) and let me know if that fixes the issue

Perfect!

**Review by ilmotta — APPROVED** [review]

**Comment by pavloburykh** [comment]
Hello @Omar! I was wondering if you could kindly clarify why the manual testing phase was skipped before merging this PR? Thank you for your assistance.

**Comment by OmarBasem** [comment]
Hi @pavloburykh, the new composer is still a work in progress and is still disabled on `develop`. There are 2 more issues remaining (https://github.com/status-im/status-mobile/issues/15331, https://github.com/status-im/status-mobile/issues/15747) which this one is a dependency for. We can do manual testing on the final issue (https://github.com/status-im/status-mobile/issues/15747). Thank you.

**Comment by pavloburykh** [comment]
> Hi @pavloburykh, the new composer is still a work in progress and is still disabled on `develop`. There are 2 more issues remaining (#15331, #15747) which this one is a dependency for. We can do manual testing on the final issue (#15747). Thank you.

Got it! Thanx for the explanation @OmarBasem

**Pull request merged**